content/blog/news/CloudPlugin-GSOC'20.md

```diff
@@ -0,0 +1,139 @@
+       
+1. [CloudManager](https://github.com/AcademySoftwareFoundation/OpenCue/blob/d4a2eca93d271dacb13ca2aaba9b531642c1dc6b/pycue/opencue/cloud/api.py#L73)
+
+    CloudManager class is the interface between the API and the widget. The CloudManager is also an abstract class
+    that will be implemented by each registered provider. 
+    This abstract class handles the following:
+    
+    * [connect()](https://github.com/AcademySoftwareFoundation/OpenCue/blob/d4a2eca93d271dacb13ca2aaba9b531642c1dc6b/pycue/opencue/cloud/api.py#L119):
+    Connection mechanism to the cloud provider through it's corresponding API
+    
+    * [create_managed_group()](https://github.com/AcademySoftwareFoundation/OpenCue/blob/d4a2eca93d271dacb13ca2aaba9b531642c1dc6b/pycue/opencue/cloud/api.py#L100):
+    Creation of a cloud pool group (CloudInstanceGroup in the API terminology)
+    
+    * [get_all_groups()](https://github.com/AcademySoftwareFoundation/OpenCue/blob/d4a2eca93d271dacb13ca2aaba9b531642c1dc6b/pycue/opencue/cloud/api.py#L112): 
+    Lists all the cloud pool groups that are available in the cloud provider's project
+    
+    `Google Cloud Platform` implementation of this class can be found here : [GoogleCloudManager](https://github.com/AcademySoftwareFoundation/OpenCue/blob/d4a2eca93d271dacb13ca2aaba9b531642c1dc6b/pycue/opencue/cloud/gce_api.py#L115)
+
+2. [CloudInstanceGroup](https://github.com/AcademySoftwareFoundation/OpenCue/blob/d4a2eca93d271dacb13ca2aaba9b531642c1dc6b/pycue/opencue/cloud/api.py#L18)
+
+    CloudInstanceGroup abstract class the container that handles the cloud provider's managed group entity. 
+    This takes care of the following:
+    
+    * Handles setting important attributes like name, id, group's instances and the status of the cloud group
+    
+    * [resize()](https://github.com/AcademySoftwareFoundation/OpenCue/blob/d4a2eca93d271dacb13ca2aaba9b531642c1dc6b/pycue/opencue/cloud/api.py#L31): 
+    Resizing the group (scaling up/down). Currently it's left to the cloud provider to take care of which instances to 
+    remove during the process of scaling down
+    
+    * [delete_cloud_group()](https://github.com/AcademySoftwareFoundation/OpenCue/blob/d4a2eca93d271dacb13ca2aaba9b531642c1dc6b/pycue/opencue/cloud/api.py#L58):
+    Deletion of the cloud group from the cloud provider
+    
+    `Google Cloud Platform` implementation of this abstract class can be found here: [GoogleCloudGroup](https://github.com/AcademySoftwareFoundation/OpenCue/blob/d4a2eca93d271dacb13ca2aaba9b531642c1dc6b/pycue/opencue/cloud/gce_api.py#L22)
+ 
+#### Plugin widget design
+
+The Cloud Manager plugin is a plugin for CueCommander. It implements the `AbstractTreeWidget` class 
+and registers itself as a CueCommander plugin. 
+
+The interface uses the main `CloudManager` api mentioned above, as the entry point to get all cloud groups
+associated with all the registered cloud providers
+
+
+## Cloud Plugin Setup
```

Comment:
Sentence case is preferred for headers, so this would be "Cloud plugin setup" (lowercase second and third word). Take a pass through the whole page to make sure its consistent.